### ISSUE TITLE
Corruption because of setBlockIndexCandidates.empty() assertion.

### ISSUE THREAD

**Issue by gabrielnergaard** [Veil-Project/veil#405]
Running OS X Mojave, wallet version 1.15, the app frequently gets into a state (twice this week) in which simply quitting and then trying to restart results in a crash. When this happens, the only solution is to start fresh from a snapshot.

From the OS X crash reporter, this is found:

Application Specific Information:
Assertion failed: (!setBlockIndexCandidates.empty()), function PruneBlockIndexCandidates, file validation.cpp, line 3265.

I haven't found a way to reproduce the error on demand. There have, however, been other reports of this in both the Veil internal support channel, and on the public side.

**Comment by onezetty** [comment]
Same happened to me at Ubuntu 16.04 

```
➜  bin veil-qt: validation.cpp:3265: void CChainState::PruneBlockIndexCandidates(): La declaración `!setBlockIndexCandidates.empty()' no se cumple.

[1]  + 15407 abort (core dumped)  ./veil-qt

```

**Comment by presstab** [comment]
I saw this once today, was not sure if it was from me switching between different branches... Will see if I can get it to happen again and save my datadirectory with the faulty state for testing.

**Comment by CryptoMaximalist** [comment]
I have 2 data directories with this error now, one of which is my snapshot node.  Both ubuntu 18.04 x64.  Let me know on discord if I can get these to you to help @presstab

**Comment by Iconoclasta** [comment]
Likewise, compiled from master - `user@ip:~$ ./veild
veild: validation.cpp:3265: void CChainState::PruneBlockIndexCandidates(): Assertion `!setBlockIndexCandidates.empty()' failed.
Aborted (core dumped)`

**Comment by presstab** [comment]
Seems to be an issue that causes the block indexes on load to have the number of tx's zeroed, then causing issues later in the loading process to arise because of it. Have not found a solution other than running with `-reindex-chainstate`

**Comment by Elysiumcore** [comment]
The same just happened to me. A simple shut down in Mac OS Mohave. Wallet will not restart.  will have to resync chain

**Comment by veilgets** [comment]
Can this be closed? It's fixed.

**Comment by codeofalltrades** [comment]
Closing. Included in release v1.0.3.0
